sdk/loadtestservice/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/TestAsyncClient.java

```diff
@@ -420,6 +427,65 @@ public Mono<Response<BinaryData>> uploadTestFileWithResponse(
         return this.serviceClient.uploadTestFileWithResponseAsync(testId, fileId, file, requestOptions);
     }
 
+    /**
+     * Upload input file for a given test name. File size can't be more than 50 MB. Existing file with same name for the
+     * given test will be overwritten. File should be provided in the request body as multipart/form-data.
+     *
+     * <p><strong>Query Parameters</strong>
+     *
+     * <table border="1">
+     *     <caption>Query Parameters</caption>
+     *     <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     *     <tr><td>fileType</td><td>Integer</td><td>No</td><td>Integer representation of the file type (0 = JMX_FILE, 1 = USER_PROPERTIES, 2 = ADDITIONAL_ARTIFACTS).</td></tr>
+     * </table>
+     *
+     * You can add these to a request with {@link RequestOptions#addQueryParam}
+     *
+     * <p><strong>Request Body Schema</strong>
+     *
+     * <pre>{@code
+     * BinaryData
+     * }</pre>
+     *
+     * <p><strong>Response Body Schema</strong>
+     *
+     * <pre>{@code
+     * {
+     *     url: String (Optional)
+     *     fileId: String (Optional)
+     *     filename: String (Optional)
+     *     fileType: String(0/1/2) (Optional)
+     *     expireTime: OffsetDateTime (Optional)
+     *     validationStatus: String (Optional)
+     * }
+     * }</pre>
+     *
+     * @param testId Unique name for load test, must be a valid URL character ^[a-z0-9_-]*$.
+     * @param fileId Unique identifier for test file, must be a valid URL character ^[a-z0-9_-]*$.
+     * @param fileName Name of test file, must be a valid URL character ^[a-z0-9_-]*$.
+     * @param file The file to be uploaded.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws IOException thrown if the multipart/form-data body cannot be constructed from file.
+     * @return fileUrl Model along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> uploadTestFileWithResponse(
+            String testId, String fileId, String fileName, BinaryData file, RequestOptions requestOptions) throws IOException {
```

Comment:
If the content is from `BinaryData`, why user need to provide a `filename`? The data might just from memory?

---

Method signature.

I think the handwritten overload should be without the last (`RequestOptions` param).

And the sync one should have `Context` param (in place of `RequestOptions`).

Reply:
`BinaryData.fromFile` discards the filename. You can refer the "Uploading .jmx file to Load Test" section in the README sample. Is there an alternate approach to preserve filename in `BinaryData` while keeping user experience simple?

Reply:
We would require `RequestOptions` parameter in Async and Sync because `filetype` is an optional query param which needs to be passed via `RequestOptions`

Reply:
If you only support upload from file, you should use a `File` parameter, instead of `BinaryData` parameter.

---

For optional parameter, the standard "WithResponse" API would have all required/optional parameters in method signature (user can choose to use `null` if they don't want to fill).

And you can have another e.g. "uploadTestFile" method with only required parameters.

---

Ref, guideline https://azure.github.io/azure-sdk/java_introduction.html